### PR TITLE
RoR group project - Recipe app

### PR DESCRIPTION
**Pull request by Zainab-Alzubair**
### In this PR

[Kanban board](https://github.com/girma336/Recipe-app/projects/1)

**1. Login page and registration page:**

- Should be built with Devise.

**2. Food list:**

- Should display a list of food added by the logged-in user as in the wireframe 
- Should lead to a form that allows users to add new food.

**3. Recipes list:**

- Should display a list of recipes created by the logged-in user as in the wireframe.
- Should lead to recipe details.
- If the user is the owner of the recipe, should allow the user to delete it.

**4. Public recipe list:**

- Should display a list of all public recipes ordered by newest as in the wireframe.
- Should lead to recipe details.

**5. Recipe details:**

- Should display a toggle button that allows for a recipe to be made public or private.
- If the recipe is public or the user is the owner of the recipe, should display the recipe details as in the wireframe.
- If the user is the owner of the recipe, should lead to the form that allows the user to add new food.

**6. Inventories list:**

- Should display a list of inventories created by the logged-in user as in the wireframe.
- Should lead to inventory details.
- If the user is the owner of the inventory, should allow to delete it.

**7. Inventory details:**

- Should display the inventory details as in the wireframe.
- Should lead to a form that allows users to add new food.

**8. Recipe details:**

- Should have a modal to choose an inventory to create a shopping list with, as in the wireframe.
- Inventory shopping list:
- Should show the list of food that is missing by comparing the food in the recipe with the food in the inventory.
- Should count the total food items and total price of the missing food.

**Thank you**

### CODE REVIEW

File: db/schema.rb [hunk 1]
```diff
@@ -0,0 +1,89 @@
...
+  # These are extensions that must be enabled in order to support this database
+  enable_extension "plpgsql"
+
+  create_table "foods", force: :cascade do |t|
+    t.string "name"
+    t.datetime "created_at", null: false
+    t.datetime "updated_at", null: false
+    t.decimal "price", precision: 8, scale: 2
+    t.string "measurement_unit"
+  end
```
Comment (ShadyShawkat) [inline db/schema.rb:23]: - [ ] Kindly make sure that the `Foods` table has a `User` reference, as per the following [Entity Relationship Diagram](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/recipe_app_erd.png?raw=true).

File: app/views/foods/index.html.erb [hunk 2]
```diff
@@ -0,0 +1,30 @@
...
+  
+    <tbody>
+      <% @foods.each_with_index do |food, index| %>
+        <tr>
+           <th scope="row"><%= index+1 %></th>
+           <td><%= food.name %></td>
+           <td><%= food.measurement_unit %></td>
+           <td><%= food.price %></td>
+           <td class="align-middle"><%= button_to 'Delete', "foods/#{food.id}", method: :delete, data: { confirm: 'Are you sure?' }, class: "btn-all" %></td>
+        </tr>
```
Comment (ShadyShawkat) [inline app/views/foods/index.html.erb:23]: - [ ] Please ensure that only the users that create foods can delete them. Currently, any user can delete any food even if they aren't the one that created it in the first place.

**Review by ShadyShawkat — CHANGES_REQUESTED** [review]
## Hi Team, 

Good job so far! 👍🏻 
There are some issues that you still need to work on to go to the next project but you are almost there!

## To Highlight
- All linters are passing ✔️ 
- Correct use of GitFlow ✔️ 
- Descriptive commit messages ✔️ 
- Professional looking pull request ✔️ 

Image

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

File: db/schema.rb (hunk 1 above)
Comment (KDlamini) [reply db/schema.rb]: Well Resolved ✔️

**Review by KDlamini — APPROVED** [review]
### Hi team @Zainab-Alzubair, @girma336, @187jjay187  🙋‍♀️,  

Excellent work! 💯 
You have done a great job implementing all the required changes. 🎊 🥳 
You have outdone yourself with this project, I like your attention to detail in color compositions. 💯 
Your feature is complete! There is nothing else to say other than... it's time to merge it. :shipit: 
Congratulations!!!

## HIGHLIGHTS 🌟 

✔️ Good PR title and description
✔️ Beautiful UI I like your styling ability 🔥 
✔️ The application has a login and Registration page, built with the Devise gem
✔️ The application is structured according to the given [wireframe](https://github.com/microverseinc/curriculum-rails/blob/main/recipe-app/images/inventory_shopping_list.png?raw=true)
✔️ Models, controllers, views, and routes set up correctly

# Status: Approved ✔️ 

oscars-standing-ovation
 
Cheers 🥂  and Happy coding!!! 👯 
Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Remember to tag me @KDlamini  in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think the code review was unfair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**